### ISSUE TITLE
Bug: AndroidContextNotFromKoin false positive on androidApplication()

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#12]
## Summary

The `AndroidContextNotFromKoin` rule incorrectly reports violations on `androidApplication()` — a legitimate Koin DSL function used to retrieve the Android `Application` instance within module definitions.

## Current Behavior

The rule fires on `androidApplication()` calls inside Koin module definitions, treating them as violations.

```kotlin
val appModule = module {
    single { MyRepository(androidApplication()) } // ← false positive
}
```

**Detekt output:**
```
AndroidContextNotFromKoin - androidContext/androidApplication should only be called in startKoin
```

## Expected Behavior

`androidApplication()` is a first-party Koin DSL function and should **not** be flagged. It's documented in the official Koin docs: https://insert-koin.io/docs/reference/koin-android/get-instances/#dsl

## Impact

7 false positives reported in a real project using standard Koin patterns.

## Environment

- Koin-Detekt 0.4.0
- Detekt 1.23.8
